Extensions/xkit_preferences.js

```diff
@@ -97,6 +94,28 @@ XKit.extensions.xkit_preferences = new Object({
 				$(".xkit_notice_container > .tab_notice_value").html(unread_mail_count);
 				$(".xkit_notice_container").addClass("tab-notice--active");
 			}
+		};
+
+		if (XKit.page.react) {
+			this.xkit_button_observer = new MutationObserver(async mutations => {
+				const account_label_selector = `[aria-label="${await account_label}"]`;
+
+				const addedAccountButton = [...mutations]
+					.flatMap(({ addedNodes }) => [...addedNodes])
+					.filter(addedNode => addedNode instanceof Element)
+					.some(element => element.querySelector(account_label_selector));
```

Comment:
instead of iterating through every single mutation of the entire webpage to check if they've added the account button, it's probably simpler & faster to just check `document.getElementById("xkit_button")` directly every time. Even simpler would be to keep a reference around to the XKit button and then check the [`isConnected`](https://developer.mozilla.org/en-US/docs/Web/API/Node/isConnected) property of the node. Honestly, given how many hundreds of thousands of mutations happen on a given tumblr page and how many of them are completely unrelated to the XKit button, it might be better to just do `isConnected` in a setInterval.

The most "correct" version of this code would register a mutation observer for every parent element of the XKit button (there's only like 5 of them) to avoid having to use `subtree: true`, but that seems like overkill from a code complexity perspective. I think getElementById in a setInterval loop might be the simplest and most maintainable option. WDYT?

Reply:
Hm, kind of. Trouble is, if the page is in the mobile view (and the button will thus never be added), this tries to add the button in the setInterval, which takes about 8ms without redesigning the jquery to remove the `:has`, which... okay, yeah, in retrospect I should have just redesigned the entire module considering how much I actually had to move around.

Reply:
Aside: re-firing all of the JQuery to generate a new button at all is pretty silly, I guess.

I think the solution with the minimal diff is reverting all of this, then just grabbing a reference to the button element after it's inserted and re-inserting that when necessary. (That would allow for the isConnected check.)

What interval would you propose for the setInterval? Not sure there's much benefit of doing that over just an unfiltered mutation observer. It doesn't fire on every individual mutation, and logging the interval between each invocation it doesn't feel to me like there would be a significant difference, and the button would pop in in that case.

Reply:
i was thinking an interval of like 100 or 200ms. but if you think the mutation observer is fine then i guess that works. agree that we should handle the "xkit button was never inserted case" but that feels like it should be handled outside of this code. long-term would be great if we could find *somewhere* to put the xkit button in mobile view, for mobile firefox &c.

Reply:
Yeah, this code has to handle the mobile view, though; resizing the page such that you go from mobile view to desktop view is one of the things that causes the button to disappear and which this code is supposed to handle.

Unfortunately, navigation unmounts `bluespaceLayout`, which contains the header container that gets changed out when you resize the page, so just putting the mutation observer on `bluespaceLayout` with no subtree key doesn't work. In general, a highly performant way of checking if the page needs the button added in the mobile view by looking for the account button is a lot of lines of code (especially without the ability to do optional chaining with this eslint version). At that point, I guess you could just check if the header is in mobile mode by doing```document.querySelector('header').parentElement.matches(XKit.css_map.keyToCss('mobileHeader'))```.